### PR TITLE
Jac/delete command

### PR DESCRIPTION
**Pull request by jacalata**
- it wasn't using the project name to search
- it kept on looking for an item after it found something to delete

New output
=====   Username: jfitzgerald
Tableau Server Site: jac
Connecting to the server...
Succeeded
DEBUG: DeleteCommand: 41: Removing Regional '''' from the server...
DEBUG: DeleteCommand: 104: Project:None, Default
DEBUG: DeleteCommand: 57: Requesting ''[Projects] Default'' from the server...
DEBUG: DeleteCommand: 57: Requesting ''[Workbooks] Regional'' from the server...
DEBUG: DeleteCommand: 79: Deleting workbook d2cd88d2-29ba-4bb9-8688-14c9f55b1167
DEBUG: DeleteCommand: 84: Succeeded

### CODE REVIEW

File: tabcmd/commands/datasources_and_workbooks/delete_command.py [hunk 1]
```diff
@@ -31,26 +32,42 @@ def run_command(args):
...
-        except TSC.ServerResponseError as workbook_error:
-            error = workbook_error
-        try:
-            item_to_delete = DeleteCommand.get_data_source_item(logger, server, args.name)
-            item_type = "datasource"
-        except TSC.ServerResponseError as ds_error:
-            error = ds_error
-        if not item_type:
+        if args.workbook or not content_type:
+            logger.debug("Attempt as workbook")
```
Comment (mwhagedorn) [inline tabcmd/commands/datasources_and_workbooks/delete_command.py:52]: if you are going to leave these in, shouldnt they be localized?

File: tabcmd/commands/datasources_and_workbooks/delete_command.py [hunk 2]
```diff
@@ -31,26 +32,42 @@ def run_command(args):
...
+            try:
+                item_to_delete = DeleteCommand.get_workbook_item(logger, server, args.name, container)
+                content_type = "workbook"
+            except TSC.ServerResponseError as error:
+                logger.debug(error)
+        if args.datasource or not content_type:
+            logger.debug("Attempt as datasource")
```
Comment (mwhagedorn) [inline tabcmd/commands/datasources_and_workbooks/delete_command.py:59]: localized?

File: tabcmd/commands/server.py [hunk 3]
```diff
@@ -48,28 +50,36 @@ def find_user_id(logger, server, username):
...
-            logger.debug("Searching in project {}".format(container))
-            req_option.filter.add(
-                TSC.Filter(TSC.RequestOptions.Field.ParentProjectId, TSC.RequestOptions.Operator.Equals, container)
-            )
+
         all_items, pagination_item = item_endpoint.get(req_option)
         if all_items is None or all_items == []:
             raise ValueError("[" + item_type + "] " + _("errors.xmlapi.not_found"))
+        if len(all_items) == 1:
+            logger.debug("Exactly one result found")
```
Comment (mwhagedorn) [inline tabcmd/commands/server.py:66]: Suggesting deleting or localizing

File: tabcmd/commands/server.py [hunk 4]
```diff
@@ -48,28 +50,36 @@ def find_user_id(logger, server, username):
...
+            result = all_items
         if len(all_items) > 1:
-            logger.debug("{}+ items of this name were found. Returning first page.".format(len(all_items)))
-            logger.debug(all_items[0].name + ", " + all_items[1].name + ", " + all_items[2].name)
+            logger.debug(
```
Comment (mwhagedorn) [inline tabcmd/commands/server.py:69]: localize?

**Comment by bhuvneshdev** [comment]
Changes look good to me, wondering if there is an update on the comments from Mike?

**Comment by jacalata** [comment]
I began updating all the strings in this but found quite a few altogether, so I'm going to do a separate commit that should cover all strings remaining.

File: tabcmd/commands/server.py (hunk 4 above)
Comment (jacalata) [reply tabcmd/commands/server.py:69]: rolling over to https://github.com/tableau/tabcmd/issues/155

File: tabcmd/commands/server.py (hunk 3 above)
Comment (jacalata) [reply tabcmd/commands/server.py:66]: Will localize, rolling over to https://github.com/tableau/tabcmd/issues/155

File: tabcmd/commands/datasources_and_workbooks/delete_command.py (hunk 1 above)
Comment (jacalata) [reply tabcmd/commands/datasources_and_workbooks/delete_command.py:52]: Yes, they should. rolling over to https://github.com/tableau/tabcmd/issues/155

File: tabcmd/commands/datasources_and_workbooks/delete_command.py (hunk 2 above)
Comment (jacalata) [reply tabcmd/commands/datasources_and_workbooks/delete_command.py:59]: rolling over to https://github.com/tableau/tabcmd/issues/155

**Pull request merged**